internal/xdscache/v3/listener_test.go

```diff
@@ -3401,6 +3402,142 @@ func TestListenerVisit(t *testing.T) {
 				SocketOptions: envoy_v3.TCPKeepaliveSocketOptions(),
 			}),
 		},
+		"httpproxy with MaxRequestsPerConnection set in listener config": {
```

Comment:
do we have a test where we assert that when we don't set it in the config we don't set it at all?

Reply:
There are other tests (such as test with defaults and few others) which would fail as the setting I introduced is off by default.